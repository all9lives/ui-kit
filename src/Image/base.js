import { css } from 'styled-components'
import Unit from '../utils/Unit'
import Size from '../utils/Size'

export const attrs = {
  width: props => (props.width || props.size),
  height: props => (props.height || props.size)
}

export const filtered = ['fit', 'size', 'height', 'width']

const base = css`
  width: ${props => Unit.convert(Size[props.width] || props.width)};
  height: ${props => Unit.convert(Size[props.height] || props.height)};
`

export default base
