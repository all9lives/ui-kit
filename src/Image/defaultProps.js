export default {
  size: '100%',
  fit: 'contain'
}
