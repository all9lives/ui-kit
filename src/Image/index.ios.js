import Image from './index.android'

export default Image
