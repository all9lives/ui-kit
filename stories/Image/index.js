import React from 'react'
import createStory from '../../storybook/createStory'
import Image from '../../src/Image'

const sourceUrl = 'https://raw.githubusercontent.com/Fineighbor/ui-kit/master/screenshots/web.png'

createStory('Image', module)
  .add('with fit `contain`', () =>
    <Image
      fit='contain'
      src={sourceUrl}
    />
  )
  .add('with fit `cover`', () =>
    <Image
      fit='cover'
      src={sourceUrl}
    />
  )
  .add('with size `200`', () =>
    <Image
      size={200}
      src={sourceUrl}
    />
  )
  .add('with width `300`, height `200`', () =>
    <Image
      width={300}
      height={200}
      fit='cover'
      src={sourceUrl}
    />
  )
  .add('with `source`', () =>
    <Image
      source={sourceUrl}
    />
  )
