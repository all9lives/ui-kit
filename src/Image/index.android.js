import styled from 'styled-components/native'
import base, { attrs } from './base'
import defaultProps from './defaultProps'

const Image = styled.Image.attrs({
  ...attrs,
  source: props => (typeof props.src === 'string' ? { uri: props.src } : props.src)
})`
  ${base}
  resizeMode: ${props => props.fit};
`
Image.defaultProps = {
  ...defaultProps,
  src: ''
}

export default Image
