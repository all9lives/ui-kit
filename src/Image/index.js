import styled from 'styled-components'
import base, { attrs, filtered } from './base'
import defaultProps from './defaultProps'
import { filterProps, mapProps } from '../utils/Misc'

export const Base = filterProps('img', ['source', ...filtered])
Base.displayName = 'Image'

export const Styled = styled(Base).attrs(attrs)`
  ${base}
  object-fit: ${props => props.fit};
`
Styled.defaultProps = {
  ...defaultProps
}

const Image = mapProps(({ src, source, ...rest }) => ({
  ...rest,
  src: src || source
}))(Styled)

export default Image
