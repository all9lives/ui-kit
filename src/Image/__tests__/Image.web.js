/* eslint-env jest */
import 'jest-styled-components'
import React from 'react'
import { shallow } from 'enzyme'
import Image, { Base, Styled } from '../index.js'

export const diveToBase = wrapper => wrapper.find(Styled).dive().find(Base).dive()

describe('Image.web', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(
      <Image />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `src`', () => {
    const wrapper = shallow(
      <Image src='any source' />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `source`', () => {
    const wrapper = shallow(
      <Image source='any source' />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `fit:cover`', () => {
    const wrapper = shallow(
      <Image fit='cover' />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `size:100`', () => {
    const wrapper = shallow(
      <Image size={100} />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `width:100`', () => {
    const wrapper = shallow(
      <Image width={100} />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })

  it('renders a snapshot with `height:100`', () => {
    const wrapper = shallow(
      <Image height={100} />
    )
    expect(diveToBase(wrapper)).toMatchSnapshot()
  })
})
