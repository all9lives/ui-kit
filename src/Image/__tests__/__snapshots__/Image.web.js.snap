// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image.web renders a snapshot 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  object-fit: contain;
}

<img
  className="c0"
  height="100%"
  size="100%"
  width="100%"
/>
`;

exports[`Image.web renders a snapshot with \`fit:cover\` 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

<img
  className="c0"
  height="100%"
  size="100%"
  width="100%"
/>
`;

exports[`Image.web renders a snapshot with \`height:100\` 1`] = `
.c0 {
  width: 100%;
  height: 100px;
  object-fit: contain;
}

<img
  className="c0"
  height={100}
  size="100%"
  width="100%"
/>
`;

exports[`Image.web renders a snapshot with \`size:100\` 1`] = `
.c0 {
  width: 100px;
  height: 100px;
  object-fit: contain;
}

<img
  className="c0"
  height={100}
  size={100}
  width={100}
/>
`;

exports[`Image.web renders a snapshot with \`src\` 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  object-fit: contain;
}

<img
  className="c0"
  height="100%"
  size="100%"
  src="https://raw.githubusercontent.com/Fineighbor/ui-kit/master/screenshots/web.png"
  width="100%"
/>
`;

exports[`Image.web renders a snapshot with \`width:100\` 1`] = `
.c0 {
  width: 100px;
  height: 100%;
  object-fit: contain;
}

<img
  className="c0"
  height="100%"
  size="100%"
  width={100}
/>
`;
